### PR TITLE
[bitnami/elasticsearch] Add support for X-Pack security

### PR DESCRIPTION
**Pull request by migruiz4**
**Description of the change**

This PR adds support for the X-Pack security settings, expected to be included in the Elasticsearch image soon.

**Benefits**

- Adds TLS encryption for both Transport layer and REST API.
- Adds support for Elasticsearch basic authentication (basic realm).
- Adds option to enable FIPS mode (requires Gold/Platinum license).
- Supports multiple formats for TLS certificates: JKS, PKCS12, and PEM certificates. All the previous formats can be password-protected or not. 
- Support for autoGenerated certificates using Helm genCA and genSignedCert (not recommended for production environments).

**Checklist** 

- [ ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [ ] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/elasticsearch/templates/coordinating-statefulset.yaml [hunk 1]
```diff
@@ -135,9 +156,9 @@ spec:
             timeoutSeconds: {{ .Values.coordinating.startupProbe.timeoutSeconds }}
             successThreshold: {{ .Values.coordinating.startupProbe.successThreshold }}
             failureThreshold: {{ .Values.coordinating.startupProbe.failureThreshold }}
-            httpGet:
-              path: /_cluster/health?local=true
-              port: 9200
+            exec:
+              command:
+                - /opt/bitnami/scripts/elasticsearch/healthcheck.sh
```
Comment (juan131) [inline bitnami/elasticsearch/templates/coordinating-statefulset.yaml:161]: Why this change?

File: bitnami/elasticsearch/templates/serviceaccount.yaml [hunk 2]
```diff
@@ -33,3 +33,12 @@ metadata:
...
 {{- end }}
+{{- if and .Values.ingest.enabled .Values.ingest.serviceAccount.create }}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "elasticsearch.ingest.serviceAccountName" . }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    role: ingest
+{{- end }}
```
Comment (juan131) [inline bitnami/elasticsearch/templates/serviceaccount.yaml:44]: Fix the warning

File: bitnami/elasticsearch/templates/ingest-statefulset.yaml [hunk 3]
```diff
@@ -1,17 +1,24 @@
...
   labels: {{- include "common.labels.standard" . | nindent 4 }}
     app.kubernetes.io/component: ingest
     ## Istio Labels: https://istio.io/docs/ops/deployment/requirements/
     app: ingest
 spec:
+  updateStrategy:
+    type: {{ .Values.master.updateStrategy.type }}
+    {{- if (eq "OnDelete" .Values.master.updateStrategy.type) }}
+    rollingUpdate: null
+    {{- end }}
```
Comment (juan131) [inline bitnami/elasticsearch/templates/ingest-statefulset.yaml:15]: Master? Please use its own values

File: bitnami/elasticsearch/values.yaml [hunk 4]
```diff
@@ -79,6 +79,75 @@ image:
   ## Set to true if you would like to see extra information on logs
   ##
   debug: false
+
```
Comment (juan131) [inline bitnami/elasticsearch/values.yaml:82]: Review that ingest and coordinating values don't have references to "Deployment"

**Review by juan131 — CHANGES_REQUESTED** [review]
Please check my suggestions

File: bitnami/elasticsearch/templates/coordinating-statefulset.yaml (hunk 1 above)
Comment (migruiz4) [reply bitnami/elasticsearch/templates/coordinating-statefulset.yaml:161]: Because Probes will fail if X-Pack security is enabled. All API endpoints, including `/_cluster/health?local=true` will be password-protected and fail because of authentication failed (code 401).

The new Elasticsearch image includes a `healtcheck.sh` with the logic for the Probes as Kubernetes does not provide mechanisms to perform an `httpsGet` with authentication and using command would require the password to be provided inline (not secret) and using the env variable ELASTICSEARCH_PASSWORD doesn't work either.

**Review by juan131 — APPROVED** [review]
LGTM

File: bitnami/elasticsearch/README.md [hunk 5]
```diff
@@ -622,6 +642,12 @@ Find more information about how to deal with common errors related to Bitnami’
 
 ## Upgrading
 
+### To 16.0.0
+
+This version replaces the Ingest and Coordinating Deployments with Statefulsets.
+
+Additionally, this version adds support for X-Pack Security features such as TLS/SSL encryption and basic authentication.
+
 ### To 15.0.0
```
Comment (juan131) [inline bitnami/elasticsearch/README.md:651]: I'm still missing instructions to upgrade from `15.x.x`

**Review by juan131 — CHANGES_REQUESTED** [review]
Sorry, I just saw this

**Review by juan131 — APPROVED** [review]
LGTM

File: bitnami/elasticsearch/values.yaml (hunk 4 above)
Comment (migruiz4) [reply bitnami/elasticsearch/values.yaml:82]: Fixed a single reference to Deployment instead of Stateful.
